### ISSUE TITLE
Data missing when reading from channels

### ISSUE THREAD

**Issue by mwane1989** [mikeobrien/TDMSReader#12]
Hi,
I have a group (mw_sensor) of close to 900 channels with each channel containing 396 data points, I am able to read from the first channel ("Untitled") fine, but when trying to read the others I'm only getting the first 12 data points.

The file is the same one I posted in a other Issue from _"System.ArgumentException" (An item with the same key has already been added)_ ..(Recordings.zip), I'm getting an error when trying to upload it.

Group: mw_sensor 
Channel: Untitled (396 data points of type System.Double)
Channel: Untitled 1 (12 data points of type System.Double)
Channel: Untitled 2 (12 data points of type System.Double)
Channel: Untitled 3 (12 data points of type System.Double)

 Any ideas on why it only reads 12 data points for all the other channels?

**Comment by innat3** [comment]
Hi, I'm writing to second this issue, as I stated in the post I wrote in stackoverflow [link](http://stackoverflow.com/questions/36313174/c-reading-tdms-files-that-possess-a-time-interval-duration) (I attached the sample file that Is giving my trouble there).

It seems that if a time interval is present in the file, when TDMSReader tries to fetch the values from the data array it takes only the amount equivalent to the "wf_samples" field instead of the "Channel Length", thus discarding/making the rest inaccessible. 

Basically, the issue is that it only reads the first interval from each channel.

**Comment by mikeobrien** [comment]
Not really sure what the problem is. Unfortunately I'm pretty slammed right now so I'd have to rely on a PR from someone to get this fixed.

**Comment by tommyja** [comment]
I am looking into it. It is probably related to incremental metadata handling.

**Comment by innat3** [comment]
thanks, let me know if you need more feedback or sample files, I'll be glad to help in any way I can

**Comment by tommyja** [comment]
PR submitted.

**Comment by mikeobrien** [comment]
Merged and pushed #13. Thx @tommyja!

**Comment by mwane1989** [comment]
Thanks @tommyja
